test/utils/widgets.dart

```diff
@@ -0,0 +1,57 @@
+/// Utility functions for testing
+
+import 'package:flutter/material.dart';
+import 'package:flutter_slidable/flutter_slidable.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:provider/provider.dart';
+
+Widget buildAppWithScreen(
+  Widget screen, [
+  List<ChangeNotifierProvider> providers = const [],
+]) {
+  final app = MultiProvider(
+    providers: providers,
+    child: MaterialApp(
+      home: Scaffold(
+        body: screen,
+      ),
+    ),
+  );
+
+  return app;
+}
+
+void expectSlideActionWith(IconData icon, String caption) {
+  final slideAction = find.byType(IconSlideAction);
+  final actionIcon = find.descendant(
+    of: slideAction,
+    matching: find.byIcon(icon),
+  );
+  final actionCaption = find.descendant(
+    of: slideAction,
+    matching: find.text(caption),
+  );
+  // ToDo: This does not ensure that the icon and the caption are from the same
+  //  slide action. The test would pass if one slideAction has the icon, but
+  // another slideAction has the caption.
+  expect(actionIcon, findsOneWidget);
+  expect(actionCaption, findsOneWidget);
+}
+
+Future<void> findAndSlideWidgetOpen(WidgetTester tester) async {
+  final slidable = find.byType(Slidable);
+  expect(slidable, findsOneWidget);
+  await slideWidgetOpen(tester, slidable);
+}
+
+Future<void> pumpAppAndSettle(WidgetTester tester, Widget app) async {
+  await tester.pumpWidget(app);
+  await tester.pumpAndSettle();
+}
+
+Future<void> slideWidgetOpen(WidgetTester tester, Finder finder) async {
+  // If this test fails unexpectedly, the slideOffset is likely wrong.
```

Comment:
Could you elaborate on your comment?

Reply:
To slide the widget open, a certain threshold of sliding has to be surpassed.
I say "certain", because in the actual _flutter_slidable_ repo they use an offset of -50 to slide the widget open (https://github.com/letsar/flutter_slidable/blob/master/test/slidable_test.dart).
For some reason this did not work for us, so I tried different (much) higher offsets to see if this changes anything.
Eventually, I settled for -500.

I find this a little fragile and actually expect there is some constant hidden somewhere.
I just couldn't find it (yet) so I added the comment in case the slideWidgetOpen tests fail at anyone elses machine.

I updated the comment to make this more clear:
```
// If this test fails unexpectedly, the slideOffset is likely not "high"
// enough to swipe the widget open.
// Increase the offset e.g. to Offset(-1000, 0) and rerun the test.
``